src/garage/_functions.py

```diff
@@ -1,10 +1,71 @@
 """Functions exposed directly in the garage namespace."""
+from collections import defaultdict
+
 from dowel import tabular
 import numpy as np
 
+import garage
 from garage.misc.tensor_utils import discount_cumsum
 
 
+def log_multitask_performance(itr, batch, discount, task_names=None):
+    """Log per-task performance and average performance.
+
+    Args:
+        itr (int): Iteration.
+        batch (TrajectoryBatch): The trajectories to evaluate with.
+        discount (float): Discount value, from algorithm's property.
+        task_names (dict[str]): Names of all tasks to evaluate.
+            In case when a task in task_names does not exist in the give batch,
+            log NaN as a placeholder.
+
+    Returns:
+        numpy.ndarray: Undiscounted returns.
+
+    """
+    traj_by_name = defaultdict(list)
+    for trajectory in batch.split():
+        try:
+            task_name = trajectory.env_infos['task_name'][0]
+        except KeyError:
+            task_name = trajectory.env_infos['task_id'][0]
+        traj_by_name[task_name].append(trajectory)
+    if task_names is None:
+        for (task_name, trajectories) in traj_by_name.items():
+            log_performance(itr,
+                            garage.TrajectoryBatch.concatenate(*trajectories),
+                            discount,
+                            prefix=task_name)
+    else:
+        for task_name in sorted(task_names):
+            if task_name in traj_by_name:
+                trajectories = traj_by_name[task_name]
+                log_performance(
+                    itr,
+                    garage.TrajectoryBatch.concatenate(*trajectories),
+                    discount,
+                    prefix=task_name)
+            else:
+                with tabular.prefix(task_name + '/'):
+                    tabular.record('Iteration', np.nan)
```

Comment:
can you leave an issue on dowel about handling this case better?

Reply:
Opened an issue at https://github.com/rlworkgroup/dowel/issues/45